cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/MetricsHandlers.java

```diff
@@ -0,0 +1,213 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.flume.Sink;
+import org.apache.flume.SinkProcessor;
+import org.apache.flume.SinkRunner;
+import org.apache.flume.Source;
+import org.apache.flume.SourceRunner;
+import org.apache.flume.source.http.HTTPSourceHandler;
+
+/**
+ *
+ * @author frb
+ */
+public final class MetricsHandlers {
+    
+    private static final CygnusLogger LOGGER = new CygnusLogger(MetricsHandlers.class);
+    
+    /**
+     * Constructor. Utility classes should not have a public or default constructor.
+     */
+    private MetricsHandlers() {
+    } // MetricsHandlers
+    
+    /**
+     * Handles GET /v1/admin/mergeMetrics and /admin/mergeMetrics.
+     * @param request
+     * @param response
+     * @param sources
+     * @param sinks
+     * @throws IOException
+     */
+    public static synchronized void get(HttpServletRequest request, HttpServletResponse response,
+            ImmutableMap<String, SourceRunner> sources, ImmutableMap<String, SinkRunner> sinks) throws IOException {
+        // Let's assume everything goes well
+        response.setStatus(HttpServletResponse.SC_OK);
+        
+        // Get the reset parameter
+        String reset = request.getParameter("reset");
+        
+        if (reset != null && !reset.equals("true") && !reset.equals("false")) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        } // if
+        
+        if (reset != null && reset.equals("true")) {
+            CygnusMetrics metrics = mergeMetrics(sources, sinks);
+            response.getWriter().println(metrics.toJsonString());
+            deleteMetrics(sources, sinks);
+            response.setContentType("application/json; charset=utf-8");
+        } else {
+            CygnusMetrics metrics = mergeMetrics(sources, sinks);
+            response.getWriter().println(metrics.toJsonString());
+            response.setContentType("application/json; charset=utf-8");
+        } // else
+    } // getServiceSubserviceMetrics
+    
+    /**
+     * Handles DELETE /v1/admin/mergeMetrics and /admin/mergeMetrics.
+     * @param response
+     * @param sources
+     * @param sinks
+     * @throws java.io.IOException
+     */
+    public static synchronized void delete(HttpServletResponse response, ImmutableMap<String, SourceRunner> sources,
+            ImmutableMap<String, SinkRunner> sinks) throws IOException {
+        deleteMetrics(sources, sinks);
+        response.setStatus(HttpServletResponse.SC_OK);
+    } // delete
+    
+    /**
+     * Merges metrics from all the given sources and sinks. It is protected in order it can be tested.
+     * @param sources
+     * @param sinks
+     * @return
+     */
+    protected static CygnusMetrics mergeMetrics(ImmutableMap<String, SourceRunner> sources,
+            ImmutableMap<String, SinkRunner> sinks) {
+        CygnusMetrics mergeServices = new CygnusMetrics();
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1383/commits/1ca87236b3c2b0b44789f1a3d2cf6801be79fb8d